packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -925,26 +917,29 @@ async fn directory_tree_to_loader_tree(
         }
 
         if let Some(subtree) = subtree {
+            let subtree = &*subtree.await?;
+
             if let Some(key) = parallel_route_key {
-                tree.parallel_routes.insert(key.into(), subtree);
+                tree.parallel_routes.insert(key.into(), subtree.clone());
                 continue;
             }
 
             // skip groups which don't have a page match.
-            if is_group_route(subdir_name) && !*subtree.has_page().await? {
+            if is_group_route(subdir_name) && !subtree.has_page() {
                 continue;
             }
 
-            if *subtree.has_page().await? {
+            if subtree.has_page() {
                 check_duplicate(&mut duplicate, subtree, app_dir).await?;
             }
 
             if let Some(current_tree) = tree.parallel_routes.get("children") {
-                if is_current_directory_catchall && *subtree.has_only_catchall().await? {
+                if is_current_directory_catchall && subtree.has_only_catchall() {
                     // there's probably already a more specific page in the
                     // slot.
-                } else if *current_tree.has_only_catchall().await? {
-                    tree.parallel_routes.insert("children".into(), subtree);
+                } else if current_tree.has_only_catchall() {
+                    tree.parallel_routes
+                        .insert("children".into(), subtree.clone());
                 } else {
```

Comment:
Doesn't looks like a good idea to me to store `parallel_routes: IndexMap<RcStr, LoaderTree>` instead of `parallel_routes: IndexMap<RcStr, Vc<LoaderTree>>` in the `LoaderTree` since it requires to clone the whole loader tree at every level?